src/rtcm3.c

```diff
@@ -685,33 +685,34 @@ static int decode_type1012(rtcm_t *rtcm)
         ppr2 =getbits(rtcm->buff,i,20); i+=20;
         lock2=getbitu(rtcm->buff,i, 7); i+= 7;
         cnr2 =getbitu(rtcm->buff,i, 8); i+= 8;
+        if (prn<40) {
+            sys=SYS_GPS;
+        }
+        else {
+            sys=SYS_SBS; prn+=80;
+        }
         if (!(sat=satno(sys,prn))) {
             trace(2,"rtcm3 1012 satellite number error: sys=%d prn=%d\n",sys,prn);
             continue;
         }
-        if (!rtcm->nav.glo_fcn[prn-1]) {
-            rtcm->nav.glo_fcn[prn-1]=fcn-7+8; /* fcn+8 */
-        }
```

Comment:
Is there a need to remove this? Could it just be qualified within `if (sys==SYS_GLO) ...`? Might also be relevant to 1010?